# Milk Data Classification

### 1. Loading the Dataset

In [65]:
# Imports
import pandas as pd
import altair as alt
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

In [56]:
milk_data = pd.read_csv('milknew.csv')
milk_data.shape

(1059, 8)

Based on the shape of the data, there are 1059 obersvations with 8 predictors

### 2. Data Preprocessing

In [57]:
milk_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1059 entries, 0 to 1058
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   pH          1059 non-null   float64
 1   Temprature  1059 non-null   int64  
 2   Taste       1059 non-null   int64  
 3   Odor        1059 non-null   int64  
 4   Fat         1059 non-null   int64  
 5   Turbidity   1059 non-null   int64  
 6   Colour      1059 non-null   int64  
 7   Grade       1059 non-null   object 
dtypes: float64(1), int64(6), object(1)
memory usage: 66.3+ KB


We can see all the 8 perdictors and their data types. The dataset contains:
- 1 float column (pH)
- 6 int columns (Temperature, Taste, Odor, Fat, Turbidity, Colour, Grade)
- 1 object column (Grade)

Since 'Grade' contains string data, we should convert that to an integer type

In [58]:
milk_data['Grade'] = milk_data['Grade'].map({'low':0,'medium':1,'high':2})
milk_data.head()

pH  Temprature  Taste  Odor  Fat   Turbidity  Colour  Grade
0  6.6          35      1     0     1          0     254      2
1  6.6          36      0     1     0          1     253      2
2  8.5          70      1     1     1          1     246      0
3  9.5          34      1     1     0          1     255      0
4  6.6          37      0     0     0          0     255      1

Now we have mapped the grade as : 
- 'Low' = 0
- 'Medium' = 1
- 'High' = 2 

In [59]:
milk_data.isna().sum()

pH            0
Temprature    0
Taste         0
Odor          0
Fat           0
Turbidity     0
Colour        0
Grade         0
dtype: int64

Our data does not contain any NA values. So we do not need to handle that.

In [60]:
# Correlation
milk_data.drop('Grade', axis=1).corr()

pH  Temprature     Taste      Odor      Fat   Turbidity  \
pH          1.000000    0.244684 -0.064053 -0.081331 -0.093429   0.048384   
Temprature  0.244684    1.000000 -0.109792 -0.048870  0.024073   0.185106   
Taste      -0.064053   -0.109792  1.000000  0.017582  0.324149   0.055755   
Odor       -0.081331   -0.048870  0.017582  1.000000  0.314505   0.457935   
Fat        -0.093429    0.024073  0.324149  0.314505  1.000000   0.329264   
Turbidity   0.048384    0.185106  0.055755  0.457935  0.329264   1.000000   
Colour     -0.164565   -0.008511 -0.082654 -0.039361  0.114151   0.136436   

              Colour  
pH         -0.164565  
Temprature -0.008511  
Taste      -0.082654  
Odor       -0.039361  
Fat         0.114151  
Turbidity   0.136436  
Colour      1.000000

We can see the correlation between all the predictors in the table above.

### 3. Data Visualizations

In [61]:
chart_grade_count = alt.Chart(milk_data).mark_bar().encode(
    x=alt.X('count()'),
    y=alt.Y('Grade', sort='x')
).properties(
    height=200,
    width=600
)
chart_grade_count

alt.Chart(...)

### 4. Data Standardization and Splitting

We will start by splitting our dataset into X (predictors) and Y (response variable). 

In [70]:
X = milk_data.drop('Grade', axis=1)
y = milk_data['Grade']

- Labels in X (Predictors) : pH, Temprature, Taste, Odor, Fat, Turbidity, Colour 
- Labels in Y (Response Variable) : Grade

Now we will scale our data using StandardScaler :

In [71]:
sc = StandardScaler()
sc.fit(X)
xscaled=sc.transform(X)
# Creating a DataFrame with the scaled features
X_scaled=pd.DataFrame(data=xscaled,columns=milk_data.columns[:-1])

Splitting the data into training and testing set:

In [73]:
X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.2, random_state=42)